kernel/kernel-api/src/test/scala/io/delta/kernel/internal/SnapshotManagerSuite.scala

```diff
@@ -0,0 +1,778 @@
+
+import scala.collection.JavaConverters._
+import scala.reflect.ClassTag
+
+import org.scalatest.funsuite.AnyFunSuite
+
+import io.delta.kernel.client._
+import io.delta.kernel.internal.fs.Path
+import io.delta.kernel.internal.snapshot.{LogSegment, SnapshotManager}
+import io.delta.kernel.internal.util.FileNames
+import io.delta.kernel.utils.{CloseableIterator, FileStatus}
+import io.delta.kernel.internal.util.Utils.toCloseableIterator
+
+class SnapshotManagerSuite extends AnyFunSuite {
+  import SnapshotManagerSuite._
+
+  test("verifyDeltaVersions") {
+    // empty array
+    SnapshotManager.verifyDeltaVersions(
+      Collections.emptyList(),
+      Optional.empty(),
+      Optional.empty())
+    // contiguous versions
+    SnapshotManager.verifyDeltaVersions(
+      Arrays.asList(1, 2, 3),
+      Optional.empty(),
+      Optional.empty())
+    // contiguous versions with correct `expectedStartVersion` and `expectedStartVersion`
+    SnapshotManager.verifyDeltaVersions(
+      Arrays.asList(1, 2, 3),
+      Optional.empty(),
+      Optional.of(3))
+    SnapshotManager.verifyDeltaVersions(
+      Arrays.asList(1, 2, 3),
+      Optional.of(1),
+      Optional.empty())
+    SnapshotManager.verifyDeltaVersions(
+      Arrays.asList(1, 2, 3),
+      Optional.of(1),
+      Optional.of(3))
+    // `expectedStartVersion` or `expectedEndVersion` doesn't match
+    intercept[IllegalArgumentException] {
+      SnapshotManager.verifyDeltaVersions(
+        Arrays.asList(1, 2),
+        Optional.of(0),
+        Optional.empty())
+    }
+    intercept[IllegalArgumentException] {
+      SnapshotManager.verifyDeltaVersions(
+        Arrays.asList(1, 2),
+        Optional.empty(),
+        Optional.of(3))
+    }
+    intercept[IllegalArgumentException] {
+      SnapshotManager.verifyDeltaVersions(
+        Collections.emptyList(),
+        Optional.of(0),
+        Optional.empty())
+    }
+    intercept[IllegalArgumentException] {
+      SnapshotManager.verifyDeltaVersions(
+        Collections.emptyList(),
+        Optional.empty(),
+        Optional.of(3))
+    }
+    // non contiguous versions
+    intercept[IllegalStateException] {
+      SnapshotManager.verifyDeltaVersions(
+        Arrays.asList(1, 3),
+        Optional.empty(),
+        Optional.empty())
+    }
+    // duplicates in versions
+    intercept[IllegalStateException] {
+      SnapshotManager.verifyDeltaVersions(
+        Arrays.asList(1, 2, 2, 3),
+        Optional.empty(),
+        Optional.empty())
+    }
+    // unsorted versions
+    intercept[IllegalStateException] {
+      SnapshotManager.verifyDeltaVersions(
+        Arrays.asList(3, 2, 1),
+        Optional.empty(),
+        Optional.empty())
+    }
+  }
+
+  //////////////////////////////////////////////////////////////////////////////////
+  // getLogSegmentForVersion tests
+  //////////////////////////////////////////////////////////////////////////////////
+
+  /* ------------------HELPER METHODS------------------ */
```

Comment:
are these helper methods ONLY for `getLogSegmentForVersion tests`? Or, will they be helpful for other tests? If so, they shouldn't be under the big heading

```
  //////////////////////////////////////////////////////////////////////////////////
  // getLogSegmentForVersion tests
  //////////////////////////////////////////////////////////////////////////////////
```

Reply:
The first 3 could be useful for other file listing mockings. They could probably just be in the companion object instead. Do you want me to move them or are you making changes here?

Reply:
> The first 3 could be useful for other file listing mockings. They could probably just be in the companion object instead.

Yeah, let's do that. I'm not making changes here, just browsing the code